### ISSUE TITLE
Explore failing //lte/gateway/c/core/oai/test/mme_app_task:sgw_config_test

### ISSUE THREAD

**Issue by electronjoe** [magma/magma#11783]
[SGWConfigTest.TestParseHealthyConfig unit test](https://sourcegraph.com/github.com/magma/magma/-/blob/lte/gateway/c/core/oai/test/mme_app_task/test_sgw_config.cpp?L76-83&subtree=true) is failing under Bazel test - it is unclear how / if it is passing in our existing CMake driven tests (possibly it is not executing).

The failure signature is ambiguous due to the way the test is written:

```
> bazel test //lte/gateway/c/core/oai/test/mme_app_task:sgw_config_test --test_output=errors
...
[ RUN      ] SGWConfigTest.TestParseHealthyConfig
unknown file: Failure
C++ exception with description "basic_string::_M_construct null not valid" thrown in the test body.
[  FAILED  ] SGWConfigTest.TestParseHealthyConfig (1 ms)
```

The error message is challenging because an exception is thrown. An exception is thrown because we use `bdata()` and do not inspect for null return value (which seems to be what is happening here) - which then causes `std::string` to throw. See test below.

```cpp
TEST(SGWConfigTest, TestParseHealthyConfig) {
  sgw_config_t sgw_config = {0};
  EXPECT_EQ(sgw_config_parse_string(sHealthyConfig, &sgw_config), 0);
  EXPECT_EQ(std::string(bdata(sgw_config.ovs_config.bridge_name)), "gtp_br0");
  EXPECT_EQ(std::string(bdata(sgw_config.ipv6.if_name_S1u_S12_S4_up)), "eth1");
  ASSERT_EQ(sgw_config.ipv6.s1_ipv6_enabled, true);
  free_sgw_config(&sgw_config);
}
```

Secondarily, the `ASSERT_EQ` should be `EXPECT_EQ` while we are here.

## Reproduction

- If you have access to GitHub Codespaces
- Go over to my [Bazel prototype branch](https://github.com/electronjoe/magma/tree/pr-giant-bazel-mme-rebase) (~4 PRs on master today)
- Click on the Green colored `Code` drop down button
- Select the `Codespaces` toggle
- Select `New Codespace` and when asked select 16 core (why not!)
- This will spin up a GitHub codespace that contains my branch
- Once the codespace is up (vs code in browswer) in the terminal window type `bazel test //lte/gateway/c/core/oai/test/mme_app_task:sgw_config_test --test_output=errors` and hit enter. This triggers a bazel build of all necessary components to run the mme_app_task:sgw_config_test, then runs the test

**Comment by koolzz** [comment]
Strange, It is executing in normal make test_oai as if I change the comparison arg it throws an error.

**Comment by electronjoe** [comment]
To clarify, @koolzz notes that in master and with `make test_oai` the test is executing and successfully.  This makes me wonder if one of our precompiler defines in Bazel at this point are causing different behavior? I'll dig.

**Comment by electronjoe** [comment]
I must say I'm scratching my head on this one.

From my gdb experiment below, it sure appears that the bstring is set to "br-br0" in `config_pP->ovs_config.bridge_name` at the return statement of `sgw_config_parse_string`. Yet after returning to our unit test the struct field seems to be `NULL`. Leaving me wondering if somehow `sgw_config_parse_string` is operating on the wrong `sgw_config_t` struct pointer?! But that cannot be because if I comment out just one line in the test - everything else passes (the issue seems very specific to the `sgw_config.ovs_config` sub-struct):

```
  // EXPECT_EQ(std::string(bdata(sgw_config.ovs_config.bridge_name)),
  // "gtp_br0");
  EXPECT_EQ(std::string(bdata(sgw_config.ipv6.if_name_S1u_S12_S4_up)), "eth1");
```

## gdb Findings 1

```
@electronjoe ➜ /workspaces/magma (pr-giant-bazel-mme-rebase ✗) $ bazel build //lte/gateway/c/core/oai/test/mme_app_task:sgw_config_test
@electronjoe ➜ /workspaces/magma (pr-giant-bazel-mme-rebase ✗) $ gdb bazel-bin/lte/gateway/c/core/oai/test/mme_app_task/sgw_config_test
Reading symbols from bazel-bin/lte/gateway/c/core/oai/test/mme_app_task/sgw_config_test...
(gdb) break sgw_config.c:394
No source file named sgw_config.c.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (sgw_config.c:394) pending.
(gdb) break sgw_config.c:449
No source file named sgw_config.c.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 2 (sgw_config.c:449) pending.
(gdb) run
...

Breakpoint 1, sgw_config_parse_string (
    config_string=0x55f708808370 "S-GW :\n{\n    NETWORK_INTERFACES :\n    {\n        # S-GW binded interface for S11 communication (GTPV2-C), if none selected the ITTI message interface is used\n        SGW_INTERFACE_NAME_FOR_S11         "..., config_pP=0x7ffd177850d0) at lte/gateway/c/core/oai/tasks/sgw/sgw_config.c:394
394           config_pP->ovs_config.bridge_name = bfromcstr(ovs_bridge_name);
(gdb) print ovs_bridge_name
$1 = 0x55f709cc0250 "gtp_br0"
(gdb) print *config_pP->ovs_config.bridge_name
Cannot access memory at address 0x0
(gdb) next
395           config_pP->ovs_config.gtp_port_num = gtp_port_num;
(gdb) print *config_pP->ovs_config.bridge_name
$2 = {mlen = 16, slen = 7, data = 0x55f709cc3300 "gtp_br0"}
(gdb) continue
Continuing.

Breakpoint 2, sgw_config_parse_string (
    config_string=0x55f708808370 "S-GW :\n{\n    NETWORK_INTERFACES :\n    {\n        # S-GW binded interface for S11 communication (GTPV2-C), if none selected the ITTI message interface is used\n        SGW_INTERFACE_NAME_FOR_S11         "..., config_pP=0x7ffd177850d0) at lte/gateway/c/core/oai/tasks/sgw/sgw_config.c:449
449       return RETURNok;
(gdb) print *config_pP->ovs_config.bridge_name
$3 = {mlen = 16, slen = 7, data = 0x55f709cc3300 "gtp_br0"}
...
(gdb) 
magma::lte::SGWConfigTest_TestParseHealthyConfig_Test::TestBody (this=0x55f709cbfe30) at lte/gateway/c/core/oai/test/mme_app_task/test_sgw_config.cpp:79
79        EXPECT_EQ(std::string(bdata(sgw_config.ovs_config.bridge_name)), "gtp_br0");
(gdb) print *sgw_config.ovs_config.bridge_name
Cannot access memory at address 0x0
```

## gdb findings 2

If I set a watchpoint on the unit test's `sgw_config.ovs_config.bridge_name` memory location, I get NO modifications before teardown of the function via failure of the `EXPECT_EQ(std::string(bdata(sgw_config.ovs_config.bridge_name)), "gtp_br0");`.  If I set a watchpoint within `sgw_config_parse_string` I see exactly one mutation of `config_pP->ovs_config.bridge_name` and it is set to a bstring with value `br-br0`. So again it's almost like the struct pointer being operated on by the function is not the correct struct pointer (HOW IS THIS POSSIBLE).

```
@electronjoe ➜ /workspaces/magma (pr-giant-bazel-mme-rebase ✗) $ bazel build //lte/gateway/c/core/oai/test/mme_app_task:sgw_config_test
@electronjoe ➜ /workspaces/magma (pr-giant-bazel-mme-rebase ✗) $ gdb bazel-bin/lte/gateway/c/core/oai/test/mme_app_task/sgw_config_test
Reading symbols from bazel-bin/lte/gateway/c/core/oai/test/mme_app_task/sgw_config_test...
(gdb) break sgw_config.c:394
No source file named sgw_config.c.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (sgw_config.c:394) pending.
(gdb) run
[ RUN      ] SGWConfigTest.TestParseHealthyConfig

Breakpoint 1, sgw_config_parse_string (
    config_string=0x55791df05370 "S-GW :\n{\n    NETWORK_INTERFACES :\n    {\n        # S-GW binded interface for S11 communication (GTPV2-C), if none selected the ITTI message interface is used\n        SGW_INTERFACE_NAME_FOR_S11         "..., config_pP=0x7ffcccfe8060) at lte/gateway/c/core/oai/tasks/sgw/sgw_config.c:394
394           config_pP->ovs_config.bridge_name = bfromcstr(ovs_bridge_name);
(gdb) watch config_pP->ovs_config.bridge_name
Hardware watchpoint 2: config_pP->ovs_config.bridge_name
(gdb) c
Continuing.

Hardware watchpoint 2: config_pP->ovs_config.bridge_name

Old value = (bstring) 0x0
New value = (bstring) 0x55791f39f750
sgw_config_parse_string (
    config_string=0x55791df05370 "S-GW :\n{\n    NETWORK_INTERFACES :\n    {\n        # S-GW binded interface for S11 communication (GTPV2-C), if none selected the ITTI message interface is used\n        SGW_INTERFACE_NAME_FOR_S11         "..., config_pP=0x7ffcccfe8060) at lte/gateway/c/core/oai/tasks/sgw/sgw_config.c:395
395           config_pP->ovs_config.gtp_port_num = gtp_port_num;
(gdb) c
Continuing.

Watchpoint 2 deleted because the program has left the block in
which its expression is valid.
0x000055791ddea29d in magma::lte::SGWConfigTest_TestParseHealthyConfig_Test::TestBody (this=0x55791f399cb0) at lte/gateway/c/core/oai/test/mme_app_task/test_sgw_config.cpp:78
warning: Source file is more recent than executable.
78        EXPECT_EQ(sgw_config_parse_string(sHealthyConfig, &sgw_config), 0);
(gdb) print sgw_config.ovs_config
$1 = {bridge_name = 0x0, gtp_port_num = 0, mtr_port_num = 0, internal_sampling_port_num = 0, internal_sampling_fwd_tbl_num = 0, uplink_port_num = 0, uplink_mac = 0x0, multi_tunnel = false, 
  gtp_echo = false, pipelined_managed_tbl0 = false, gtp_csum = false}
```

## To run Prototype Bazel Test

- If you have access to GitHub Codespaces (you should because you are a member of GH Magma)
- Go over to my [Bazel prototype branch](https://github.com/electronjoe/magma/tree/pr-giant-bazel-mme-rebase) (~4 PRs on master today)
- Click on the Green colored `Code` drop down button
- Select the `Codespaces` toggle
- Select `New Codespace` and when asked select 16 core (why not!)
- This will spin up a GitHub codespace that contains my branch
- Once the codespace is up (vs code in browswer) in the terminal window type `bazel test //lte/gateway/c/core/oai/test/mme_app_task:sgw_config_test --test_output=errors` and hit enter. This triggers a bazel build of all necessary components to run the ngap_state_converter_test, then runs the test

**Comment by electronjoe** [comment]
Resolved by #12018
